examples/rl/DDPG_Pendulum.py

```diff
@@ -0,0 +1,428 @@
+
+            critic_loss = tf.math.reduce_mean(tf.math.square(y - critic_value))
+
+        critic_grad = tape.gradient(critic_loss, critic_model.trainable_variables)
+        critic_optimizer.apply_gradients(
+            zip(critic_grad, critic_model.trainable_variables)
+        )
+
+        with tf.GradientTape() as tape:
+
+            actor_part = actor_model(state_batch)
+            critic_part = critic_model([state_batch, actor_part])
+            # used -ve as we want to max the value given by critic on our actions
+            actor_loss = -tf.math.reduce_mean(critic_part)
+
+        actor_grad = tape.gradient(actor_loss, actor_model.trainable_variables)
+        actor_optimizer.apply_gradients(
+            zip(actor_grad, actor_model.trainable_variables)
+        )
+
+
+# This update target Parameters slowly
+#  on basis of tau that is much less than one.
+def update_target(tau):
+
+    for i, target_layer in enumerate(target_critic.layers):
+
+        layer = critic_model.get_layer(index=i)
+
+        if target_layer.weights != []:
+
+            layer_var = layer.get_weights()
+            target_var = target_layer.get_weights()
+
+            new_W = layer_var[0] * tau + target_var[0] * (1 - tau)
+            new_b = layer_var[1] * tau + target_var[1] * (1 - tau)
+
+            target_layer.set_weights([new_W, new_b])
+
+    for i, target_layer in enumerate(target_actor.layers):
+
+        layer = actor_model.get_layer(index=i)
+
+        if target_layer.weights != []:
+
+            layer_var = layer.get_weights()
+            target_var = target_layer.get_weights()
+
+            new_W = layer_var[0] * tau + target_var[0] * (1 - tau)
+            new_b = layer_var[1] * tau + target_var[1] * (1 - tau)
+
+            target_layer.set_weights([new_W, new_b])
+
+
+"""
+Here we declare Actor and Critic Networks. Its simple Deep Learning 101. Basic Dense
+layers with 'ReLU' Activation.
+
+NOTICE : We use initialization for last layer of actor to be between -0.003 to 0.003 as
+this prevents from reaching 1 or -1 value in initial stages which will make our make our
+Gradient Zero Because its "tanh"!!
+"""
+
+
+def get_actor():
+
+    init = tf.keras.initializers.GlorotNormal(seed=1)
+    last_init = tf.random_uniform_initializer(minval=-0.003, maxval=0.003)
+
+    inputs = layers.Input(shape=(nS))
+    out = layers.Dense(512, activation="relu", kernel_initializer=init)(inputs)
+    out = layers.LayerNormalization()(out)
+    out = layers.Dense(512, activation="relu", kernel_initializer=init)(out)
+    out = layers.LayerNormalization()(out)
+    outputs = layers.Dense(1, activation="tanh", kernel_initializer=last_init)(out)
+
+    # our upper bound is 2.0 for Pendulum.
+    # This scale out our Actions
+    outputs = outputs * upper_bound
+    model = tf.keras.Model(inputs, outputs)
+    return model
+
+
+def get_critic():
+
+    init = tf.keras.initializers.GlorotNormal(seed=1)
```

Comment:
You can remove the seed; the code is not deterministic in its current state. Just use the string "glorot_normal".

Question: does this initializer really help, compared to the default (uniform)?